packages/bitfield/src/lib/BitField.ts

```diff
@@ -0,0 +1,279 @@
+export class BitField<Flags extends Record<string, number> | Record<string, bigint>> {
+	public readonly type: Flags[keyof Flags] extends number ? 'number' : 'bigint';
+	public readonly zero: Flags[keyof Flags] extends number ? 0 : 0n;
+	public readonly mask: ValueType<this>;
+	public readonly flags: Flags;
+
+	public constructor(flags: Readonly<Flags>) {
+		if (typeof flags !== 'object' || flags === null) {
+			throw new TypeError('flags must be a non-null object');
+		}
+
+		const entries = Object.entries(flags) as [string, Flags[keyof Flags]][];
+		if (entries.length === 0) {
+			throw new TypeError('flags must be a non-empty object');
+		}
+
+		const type = typeof entries[0][1];
+		if (type !== 'number' && type !== 'bigint') {
+			throw new TypeError('type must be either "number" or "bigint"');
```

Comment:
```suggestion
			throw new TypeError('A bitfield can only use `number`s or `bigint`s for its values.');
```

This is all Vladdy asked. Was that so hard...